[FSDP] generic argument forward for load_local_state_dict

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #73325

Make sure we don't have to modify this function if the API signature
of load_state_dict changes, because we don't really handle these args, just
pass them on to nn.Module impl.

Differential Revision: [D34433606](https://our.internmc.facebook.com/intern/diff/D34433606/)

rohan-varma committed:
[FSDP] generic argument forward for load_local_state_dict

Make sure we don't have to modify this function if the API signature
of load_state_dict changes, because we don't really handle these args, just
pass them on to nn.Module impl.

Differential Revision: [D34433606](https://our.internmc.facebook.com/intern/diff/D34433606/)

[ghstack-poisoned]

fegin reviewed (approved):
Thanks for the change.

rohan-varma committed:
Update on "[FSDP] generic argument forward for load_local_state_dict"


Make sure we don't have to modify this function if the API signature
of load_state_dict changes, because we don't really handle these args, just
pass them on to nn.Module impl.

Differential Revision: [D34433606](https://our.internmc.facebook.com/intern/diff/D34433606/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[FSDP] generic argument forward for load_local_state_dict"


Make sure we don't have to modify this function if the API signature
of load_state_dict changes, because we don't really handle these args, just
pass them on to nn.Module impl.

Differential Revision: [D34433606](https://our.internmc.facebook.com/intern/diff/D34433606/)

[ghstack-poisoned]
